### PR TITLE
Bugfix for L1 jet sorting

### PR DESCRIPTION
**Pull request by kalderon**
As spotted by @aakilgal, when multiple L1 jets have the same ET the sorting was not filling eta and phi correctly. This sorts them properly.

### CODE REVIEW

File: Root/HelpTreeBase.cxx [hunk 1]
```diff
@@ -625,6 +625,10 @@ void HelpTreeBase::AddL1Jets()
 
 }
 
+bool sortFunction(const std::vector<float> a, const std::vector<float> b) {
```
Comment (kratsg) [inline Root/HelpTreeBase.cxx]: Please put this code into HelperFunctions (https://github.com/UCATLAS/xAODAnaHelpers/blob/master/xAODAnaHelpers/HelperFunctions.h) alongside our other sort_pt functions there.

File: Root/HelpTreeBase.cxx [hunk 2]
```diff
@@ -639,23 +643,28 @@ void HelpTreeBase::FillL1Jets( const xAOD::JetRoIContainer* jets, bool sortL1Jet
...
-
-    for( unsigned int i = 0; i < L1jet_Et.size(); i++) {
-      int index = std::find (L1jet_Et.begin(), L1jet_Et.end(), L1jet_Et_sorted.at(i)) - L1jet_Et.begin();
-      m_l1Jet_et8x8.push_back ( jets->at(index)->et8x8() / m_units );
-      m_l1Jet_eta.push_back( jets->at(index)->eta() );
-      m_l1Jet_phi.push_back( jets->at(index)->phi() );
-      m_nL1Jet++;
+    
+    std::sort(vec.begin(), vec.end(), sortFunction);
+      for (int i = int(vec.size())-1; i >= 0; i--) {
```
Comment (kratsg) [inline Root/HelpTreeBase.cxx]: fix indentation here

File: Root/HelpTreeBase.cxx [hunk 3]
```diff
@@ -639,23 +643,28 @@ void HelpTreeBase::FillL1Jets( const xAOD::JetRoIContainer* jets, bool sortL1Jet
...
+	m_l1Jet_et8x8.push_back((vec.at(i)).at(0) / m_units);
+	m_l1Jet_eta.push_back((vec.at(i)).at(1));
+	m_l1Jet_phi.push_back((vec.at(i)).at(2));
+	m_nL1Jet++;
+      }
```
Comment (kratsg) [inline Root/HelpTreeBase.cxx]: fix indentation here and below

**Comment by kalderon** [comment]
Hi @kratsg,

@akilgall made some edits to address your requests above, and I think they passed most of the checks but for some reason the 21.2.34 builds failed. It seems this is related to METConstructor, nothing obviously linked to this pull request - see L632 onwards of https://travis-ci.org/UCATLAS/xAODAnaHelpers/jobs/451069192. Can you advise what we should do further?

Will

**Comment by kalderon** [comment]
Ah, it's because of https://github.com/UCATLAS/xAODAnaHelpers/pull/1268.  "Update MET significance interface, bump to 21.2.51 minimum". So it's expected that everything before that will fail.

**Pull request merged**